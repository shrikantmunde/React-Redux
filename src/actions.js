export const ADD_NAME = 'ADD_NAME'
export const REMOVE_NAME = 'REMOVE_NAME'
export const COPY_NAMES = 'COPY_NAMES'
export const SET_USERNAME = 'SET_USERNAME'
export const SET_PASSWORD = 'SET_PASSWORD'

export const REQUEST_LOGIN = 'REQUEST_LOGIN'
export const LOGIN_ERROR = 'LOGIN_ERROR'
export const LOGIN_OK = 'LOGIN_OK'
