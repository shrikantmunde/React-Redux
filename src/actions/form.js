import { SET_USERNAME, SET_PASSWORD } from '../actions'

export function setUsername(username) {
  return {
    type: SET_USERNAME,
    username
  }
}

export function setPassword(password) {
  return {
    type: SET_PASSWORD,
    password
  }
}
