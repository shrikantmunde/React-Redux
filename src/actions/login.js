import { REQUEST_LOGIN, LOGIN_OK, LOGIN_ERROR } from '../actions'

// function requestLogin() {
//   return {
//     type: REQUEST_LOGIN
//   }
// }

const requestLogin = () => ({ type: REQUEST_LOGIN })

function loginOk(response) {
  return {
    type: LOGIN_OK,
    response
  }
}

function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error
  }
}

export function login(username, password) {
  return function(dispatch) {
    if (!username || !password) return

    dispatch(requestLogin())

    window.fetch('https://api.ipify.org?format=json')
      .then(response => {
        if (!response.ok) throw new Error(response.status)
        return response.json()
      })
      .then(response => dispatch(loginOk(response)))
      .catch(ex => dispatch(loginError(ex.message)))
  }
}
