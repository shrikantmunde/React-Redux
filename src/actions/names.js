import { ADD_NAME, REMOVE_NAME, COPY_NAMES } from '../actions'

// export const addName = (name) => {
//   return {
//     type: ADD_NAME,
//     name
//   }
// }

// This is an action creatore
// It returns a simple object that describes the change you
// want to make to the state.
// All the actions must implement the "type" property
// Use a unique ID for the type
// Most people use a string constant
export function addName(name) {
  return {
    type: ADD_NAME,
    name : name
  }
}

export function removeName(index) {
  return {
    type: REMOVE_NAME,
    index 
  }
}

export function copyNames() {
  return {
    type: COPY_NAMES
  }
}
