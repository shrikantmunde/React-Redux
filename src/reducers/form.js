import * as types from '../actions'

const initialState = {
  username: '',
  password: ''
}

function form(state = initialState, action) {
  switch (action.type) {
    case types.SET_USERNAME:
      return {
        ...state,
        username: action.username
      }
    case types.SET_PASSWORD:
      return {
        ...state,
        password: action.password
      }
    default:
      return state
  }
}

export default form

