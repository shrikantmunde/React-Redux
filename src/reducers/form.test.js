import form from './form'
import { setUsername, setPassword } from '../actions/form'

it('returns the initial state', () => {
  const state = form(undefined, {})
  expect(state).toEqual({
    username: '',
    password: ''
  })
})

it('returns the initial state', () => {
  const state = form(undefined, {})
  expect(state).toMatchSnapshot()
})

it('handles SET_USERNAME', () => {
  let state = {
    username: 'test',
    password: 'password'
  }
  state = form(state, setUsername('NEWUSER'))
  expect(state).toEqual({
    username: 'NEWUSER',
    password: 'password'
  })
  expect(state).toMatchSnapshot()
})

it('handles SET_PASSWORD', () => {
  let state = {
    username: 'username',
    password: 'old password'
  }
  state = form(state, setPassword('new password'))
  expect(state).toEqual({
    username: 'username',
    password: 'new password'
  })
  // expect(state).toMatchSnapshot()
})
