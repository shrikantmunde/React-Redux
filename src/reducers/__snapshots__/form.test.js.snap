exports[`test handles SET_USERNAME 1`] = `
Object {
  "password": "password",
  "username": "NEWUSER",
}
`;

exports[`test returns the initial state 1`] = `
Object {
  "password": "",
  "username": "",
}
`;
